core/types/rlp_backwards_compat.libevm_test.go

```diff
@@ -106,3 +109,196 @@ func testHeaderRLPBackwardsCompatibility(t *testing.T) {
 		assert.Equal(t, hdr, got)
 	})
 }
+
+func TestBodyRLPBackwardsCompatibility(t *testing.T) {
+	rng := ethtest.NewPseudoRand(0)
+
+	randTx := func() *Transaction { return NewTx(&LegacyTx{Nonce: rng.Uint64()}) }
+	randHdr := func() *Header { return &Header{ParentHash: rng.Hash()} }
+	randWithdraw := func() *Withdrawal { return &Withdrawal{Index: rng.Uint64()} }
+
+	// We build up test-case [Body] instances from the power set of each of
+	// these components.
+	txs := [][]*Transaction{
+		nil, {}, // Must be equivalent for non-optional field
+		{randTx()},
+		{randTx(), randTx()}, // Demonstrates nested lists
+	}
+	uncles := [][]*Header{
+		nil, {},
+		{randHdr()},
+		{randHdr(), randHdr()},
+	}
+	withdrawals := [][]*Withdrawal{
+		nil, {}, // Must be different for optional field
+		{randWithdraw()},
+		{randWithdraw(), randWithdraw()},
+	}
+
+	var bodies []*Body
+	for _, t := range txs {
+		for _, u := range uncles {
+			for _, w := range withdrawals {
+				bodies = append(bodies, &Body{t, u, w})
+			}
+		}
+	}
+
+	for _, body := range bodies {
+		t.Run("", func(t *testing.T) {
+			t.Logf("%T: %+[1]v", body)
```

Comment:
that should do I think?

```suggestion
		name := fmt.Sprintf("%T: %+[1]v", body)
		t.Run(name, func(t *testing.T) {
```

Reply:
Making it the name will mangle the struct printing.